### ISSUE TITLE
БЭМ

### ISSUE THREAD

**Issue by Hahahayabanan** [ivanushkaPr/toxin-refactoring#6]
Что такое `_form`? Если это имя блока, то почему использовано нижнее подчёркивание, а не дефис?
![image](https://user-images.githubusercontent.com/43137324/84180350-6be2d480-aa98-11ea-8c99-2b3d0a08e424.png)

**Comment by Hahahayabanan** [comment]
По стандартам п.1 https://github.com/fullstack-development/front-end-best-practices/blob/master/CSS/bem.md

> блок, элемент — всегда имя существительное (noun);

![image](https://user-images.githubusercontent.com/43137324/84181611-38a14500-aa9a-11ea-83fd-439111946f3d.png)

**Comment by Hahahayabanan** [comment]
Элемент модификатора элемента?

![image](https://user-images.githubusercontent.com/43137324/84182447-7783ca80-aa9b-11ea-9ef6-0bcea0c07309.png)

**Comment by Hahahayabanan** [comment]
Не стоит использовать такие длинные имена нигде

Модификатор size, судя по имени, должен быть булевым, тогда значение должно отделяться от имени двумя дефисами.

> block-name__elem-name--mod-name--mod-val

![image](https://user-images.githubusercontent.com/43137324/84181716-61293f00-aa9a-11ea-9325-41a9104daaf0.png)

**Comment by Hahahayabanan** [comment]
А вообще, рекомендую тебе использовать стандартную схему именования, с ней должно быть меньше путаницы

**Comment by ivanushkaPr** [comment]
> По стандартам п.1 https://github.com/fullstack-development/front-end-best-practices/blob/master/CSS/bem.md
> 
> > блок, элемент — всегда имя существительное (noun);
> 
> ![image](https://user-images.githubusercontent.com/43137324/84181611-38a14500-aa9a-11ea-83fd-439111946f3d.png)

Исправил.

**Comment by ivanushkaPr** [comment]
> Не стоит использовать такие длинные имена нигде
> 
> Модификатор size, судя по имени, должен быть булевым, тогда значение должно отделяться от имени двумя дефисами.
> 
> > block-name__elem-name--mod-name--mod-val
> 
> ![image](https://user-images.githubusercontent.com/43137324/84181716-61293f00-aa9a-11ea-9325-41a9104daaf0.png)



> Не стоит использовать такие длинные имена нигде
> 
> Модификатор size, судя по имени, должен быть булевым, тогда значение должно отделяться от имени двумя дефисами.
> 
> > block-name__elem-name--mod-name--mod-val
> 
> ![image](https://user-images.githubusercontent.com/43137324/84181716-61293f00-aa9a-11ea-9325-41a9104daaf0.png)

Во время изменения правил именования по БЭМ, я сверился с информацией на сайте: https://ru.bem.info/methodology/naming-convention/

Не нашел там информации о необходимости разделения модификаторов ключ - значения(как я понимаю вы их имели ввиду, а не булевые) двумя дефисами, там пишут про один дефис.

В разделе о best practice тоже не нашел информации о именование.
В любом случае, все модификаторы типа ключ - значение были разделены дефисом.

**Comment by Hahahayabanan** [comment]
> Не нашел там информации о необходимости разделения модификаторов ключ - значения(как я понимаю вы их имели ввиду, а не булевые) двумя дефисами при классической схеме наименования, там пишут про один один дефис.

Да, ключ-значение, конечно же. Так как у тебя в проекте использовался альтернативный стил, я для него и указал) 
Вот это имел ввиду
![image](https://user-images.githubusercontent.com/43137324/84854581-0cc22880-b073-11ea-8654-fe22f6e913e8.png)

Классический в FSD
![image](https://user-images.githubusercontent.com/43137324/84855587-7e9b7180-b075-11ea-8595-ab60c66d9c2c.png)

**Comment by Hahahayabanan** [comment]
По стандартам п.1 https://github.com/fullstack-development/front-end-best-practices/blob/master/CSS/bem.md

> модификатор — должен удовлетворять свойствам модификатора из английского языка. Всегда adjective или adjectival phrase

![image](https://user-images.githubusercontent.com/43137324/84856650-54977e80-b078-11ea-9b68-40236e91c233.png)
![image](https://user-images.githubusercontent.com/43137324/84856673-6416c780-b078-11ea-9fa2-b4d70f99feb3.png)

**Comment by ivanushkaPr** [comment]
> > Не нашел там информации о необходимости разделения модификаторов ключ - значения(как я понимаю вы их имели ввиду, а не булевые) двумя дефисами при классической схеме наименования, там пишут про один один дефис.
> 
> Да, ключ-значение, конечно же. Так как у тебя в проекте использовался альтернативный стил, я для него и указал)
> Вот это имел ввиду
> ![image](https://user-images.githubusercontent.com/43137324/84854581-0cc22880-b073-11ea-8654-fe22f6e913e8.png)
> 
> Классический в FSD
> ![image](https://user-images.githubusercontent.com/43137324/84855587-7e9b7180-b075-11ea-8595-ab60c66d9c2c.png)

Понял, спасибо. Выходит я сам и себя, и тебя запутал =)
